utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -36,14 +43,37 @@ public class NonBlocking implements CachedSupplier.PrefetchStrategy {
     /**
      * Single threaded executor to asynchronous refresh the value.
      */
-    private final ExecutorService executor;
+    private final ScheduledExecutorService executor;
 
     /**
      * Create a non-blocking prefetch strategy that uses the provided value for the name of the background thread that will be
      * performing the update.
      */
     public NonBlocking(String asyncThreadName) {
-        this.executor = ExecutorUtils.newSingleDaemonThreadExecutor(1, asyncThreadName);
+        this.executor = newExecutor(asyncThreadName);
+    }
+
+    private static ScheduledExecutorService newExecutor(String asyncThreadName) {
+        Validate.paramNotBlank(asyncThreadName, "asyncThreadName");
+        return new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().daemonThreads(true)
+                                                                            .threadNamePrefix(asyncThreadName)
+                                                                            .build());
+    }
+
+    @Override
+    public void initializeCachedSupplier(CachedSupplier<?> cachedSupplier) {
+        executor.execute(cachedSupplier::get);
+        scheduleRefresh(cachedSupplier::get);
+    }
+
+    private void scheduleRefresh(Runnable valueUpdater) {
+        executor.schedule(() -> {
+            try {
+                valueUpdater.run();
+            } finally {
+                scheduleRefresh(valueUpdater);
+            }
+        }, 1, MINUTES);
```

Comment:
Can we make this scheduled wait time in minutes as configurable at the time of class initiation and provide default value of 1 minutes ?

Reply:
I considered making it configurable, but I didn't see a compelling reason to do so, since it's not a cache busting refresh. It just makes sure the cache doesn't go too cold. Functionally, configuring this to 1 second wouldn't keep the cache much fresher, and configuring it to 30 minutes doesn't really save much performance.

Reply:
+1, seems reasonable